packages/styles/src/private/makeCreateStyles.ts

```diff
@@ -0,0 +1,15 @@
+function createStyle(content: string) {
+  const style = document.createElement('style');
+  style.append(document.createTextNode(content));
+  return style;
+}
+
+export default function makeCreateStyles(...contents: string[]) {
+  return function createStyles() {
+    if (!globalThis.document) {
```

Comment:
Should we throw if `document` is not there? I know SSR, but this function seems like it will create (detached) styles.

Could also be, `safeMakeCreateStyles`, `makeCreateSafeStyles`, `safeMakeCreateDetachedStyles`...